lib/linguist/languages.yml

```diff
@@ -8412,3 +8412,19 @@ xBase:
   tm_scope: source.harbour
   ace_mode: text
   language_id: 421
+TypeSpec:
+  type: programming
+  color: "#4A3665"
+  aliases:
+  - tsp
+  extensions:
+  - ".tsp"
+  tm_scope: source.tsp
+  ace_mode: text
+  language_id: 952272597
+TravelingSalesmanProblem:
```

Comment:
A more appropriate name for this language would be `TSPLIB data`, since—according to [its specification](http://comopt.ifi.uni-heidelberg.de/software/TSPLIB95/tsp95.pdf#page=2)—the file format spans optimisation problems other than the [TSP](https://en.wikipedia.org/wiki/Travelling_salesman_problem):

> ### 1.1.2 <samp>TYPE</samp> : &lt;<var>string</var>&gt;
> Specifies the type of the data. Possible types are
> <pre><b>TSP</b>  Data for a symmetric travelling salesman problem
> <b>ATSP</b> Data for an asymmetric travelling salesman problem
> <b>SOP</b>  Data for a sequential ordering problem
> <b>HCP</b>  Hamiltonian cycle problem data
> <b>CVRP</b> Capacitated vehicle routing problem data
> <b>TOUR</b> A collection of tours</pre>

`TSPLIB` is the name of the data sample collection that uses the `.tsp` extension, but I'd nonetheless add an `aliases` field with the problem's full names in English (<samp>en-GB</samp>) and Yanklish (<samp>en-US</samp>), respectively:

```suggestion
TSPLIB data:
  aliases:
  - travelling salesman problem
  - traveling salesman problem
```